koans/src/java7/AboutTryWithResources.java

```diff
@@ -0,0 +1,106 @@
+package java7;
+
+import com.sandwich.koan.Koan;
+
+import java.io.*;
+
+import static com.sandwich.koan.constant.KoanConstants.__;
+import static com.sandwich.util.Assert.assertEquals;
+
+public class AboutTryWithResources {
+
+    @Koan
+    public void lookMaNoClose() {
+        String str = "first line"
+                + System.lineSeparator()
+                + "second line";
+        InputStream is = new ByteArrayInputStream(str.getBytes());
+        String line;
+        /* BufferedReader implementing @see java.lang.AutoCloseable interface */
+        try (BufferedReader br =
+                     new BufferedReader(
+                             new InputStreamReader(is))) {
+            line = br.readLine();
+            //br guaranteed to be closed
+        } catch (IOException e) {
+            line = "error";
+        }
+        assertEquals(line, __);
+    }
+
+    @Koan
+    public void lookMaNoCloseWithException() throws IOException {
+        String line;
+        try (BufferedReader br =
+                     new BufferedReader(
+                             new FileReader("I do not exist!"))) {
+            line = br.readLine();
+        }
+        assertEquals(line, __);
+    }
+
+    @Koan
+    public void lookMaNoCloseWithMultipleResources() throws IOException {
+        String str = "first line"
+                + System.lineSeparator()
+                + "second line";
+        InputStream is = new ByteArrayInputStream(str.getBytes());
+        String line;
+        //multiple resources in the same try declaration
+        try (BufferedReader br =
+                     new BufferedReader(
+                             new FileReader("I do not exist!"));
+             BufferedReader brFromString =
+                     new BufferedReader(
+                             new InputStreamReader(is))
+        ) {
+            line = br.readLine();
+            line += brFromString.readLine();
+        }catch (IOException e) {
+            line = "error";
+        }
+        assertEquals(line, __);
+    }
+
+    @Koan
+    public void supressException(){
```

Comment:
Wow.  I didn't realize that suppressed exceptions existed.  That's cool.  I wonder if something like `exceptionThrownFromCloseIsSuppressed` would be a more descriptive name for this Koan.